ui/base/src/test/java/org/cru/godtools/base/ui/util/ModelUtilsRobolectricTest.kt

```diff
@@ -0,0 +1,45 @@
+package org.cru.godtools.base.ui.util
+
+import android.app.Activity
+import android.content.Context
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.cru.godtools.model.Tool
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import java.util.Locale
+
+@RunWith(AndroidJUnit4::class)
+class ModelUtilsRobolectricTest {
+    private lateinit var context: Context
+    private val tool = Tool().apply {
+        category = "gospel"
+    }
+    private val toolNull: Tool? = null
+
+    @Before
+    fun setup() {
+        context = Robolectric.buildActivity(Activity::class.java).get()
+    }
+
+    @Test
+    fun testToolGetCategoryNullTool() {
+        assertEquals("", toolNull.getCategory(context, null))
+        assertEquals("", toolNull.getCategory(context, Locale.FRENCH))
+    }
+
+    @Test
+    fun testToolGetCategoryValidCategory() {
+        assertEquals("Gospel Invitation", tool.getCategory(context, null))
+        assertEquals("Invitation Evangile", tool.getCategory(context, Locale.FRENCH))
+    }
+
+    @Test
+    fun testToolGetCategoryUnknownCategory() {
+        tool.category = "unknown"
```

Comment:
I think in the Java web space, sometimes tests get run out of order. If this test happened to get run before the one above it, wouldn't the one above it fail?

Reply:
JUnit actually runs each test it a fresh instance of the test class

Reply:
Ah, interesting.